hibernate-reactive-core/src/test/java/org/hibernate/reactive/timezones/TimeZoneStorageMappingTests.java

```diff
@@ -0,0 +1,308 @@
+							  Tuple.class
+					  ).getResultList()
+					  .thenAccept( resultList -> {
+						  assertThat( resultList.get( 0 ).get( 0, OffsetTime.class ) ).isEqualTo( OFFSET_TIME );
+						  assertThat( resultList.get( 0 ).get( 1, OffsetDateTime.class ) ).isEqualTo( OFFSET_DATE_TIME );
+						  assertThat( resultList.get( 0 ).get( 2, ZonedDateTime.class ) ).isEqualTo( ZONED_DATE_TIME );
+						  assertThat( resultList.get( 0 ).get( 3, ZoneOffset.class ) ).isEqualTo( OFFSET_TIME.getOffset() );
+						  assertThat( resultList.get( 0 ).get( 4, ZoneOffset.class ) ).isEqualTo( OFFSET_DATE_TIME.getOffset() );
+						  assertThat( resultList.get( 0 ).get( 5, ZoneOffset.class ) ).isEqualTo( ZONED_DATE_TIME.getOffset() );
+						  assertThat( resultList.get( 0 ).get( 6, OffsetTime.class ) ).isEqualTo( OFFSET_TIME.plusHours( 1L ) );
+						  assertThat( resultList.get( 0 ).get( 7, OffsetDateTime.class ) ).isEqualTo( OFFSET_DATE_TIME.plusHours( 1L ) );
+						  assertThat( resultList.get( 0 ).get( 8, ZonedDateTime.class ) ).isEqualTo( ZONED_DATE_TIME.plusHours( 1L ) );
+						  assertThat( resultList.get( 0 ).get( 9, Duration.class ) ).isEqualTo( Duration.ofHours( 1L ) );
+						  assertThat( resultList.get( 0 ).get( 10, Duration.class ) ).isEqualTo( Duration.ofHours( 1L ) );
+						  assertThat( resultList.get( 0 ).get( 11, Duration.class ) ).isEqualTo( Duration.ofHours( 1L ) );
+					  } )
+				)
+		);
+	}
+
+	public void testOffsetRetainedFormat(VertxTestContext context, String suffix) {
+		test( context, openSession()
+				.thenCompose( session -> session.createQuery(
+							  "select " +
+									  "format(e.offsetTime" + suffix + " as 'HH:mm:ssxxx'), " +
+									  "format(e.offsetDateTime" + suffix + " as 'dd/MM/yyyy ''at'' HH:mm:ssxxx'), " +
+									  "format(e.zonedDateTime" + suffix + " as 'dd/MM/yyyy ''at'' HH:mm:ssxxx'), " +
+									  "1 from TimeZoneStorageEntity e " +
+									  "where e.offsetDateTime" + suffix + " = e.offsetDateTime" + suffix,
+							  Tuple.class
+					  ).getResultList()
+					  .thenAccept( resultList -> {
+						  assertThat( resultList.get( 0 ).get( 0, String.class ) ).isEqualTo( TIME_FORMATTER.format( OFFSET_TIME ) );
+						  assertThat( resultList.get( 0 ).get( 1, String.class ) ).isEqualTo( FORMATTER.format( OFFSET_DATE_TIME ) );
+						  assertThat( resultList.get( 0 ).get( 2, String.class ) ).isEqualTo( FORMATTER.format( ZONED_DATE_TIME ) );
+					  } )
+				)
+		);
+	}
+
+	@Test
+	public void testNormalize(VertxTestContext context) {
+		test( context, openSession()
+				.thenCompose( session -> session.createQuery(
+							"select " +
+									"e.offsetTimeNormalized, " +
+									"e.offsetDateTimeNormalized, " +
+									"e.zonedDateTimeNormalized, " +
+									"e.offsetTimeNormalizedUtc, " +
+									"e.offsetDateTimeNormalizedUtc, " +
+									"e.zonedDateTimeNormalizedUtc " +
+									"from TimeZoneStorageEntity e",
+							Tuple.class
+					).getResultList()
```

Comment:
if you use `getSingleResult()`, you can avoid having `resultList.get(0)` each time

Reply:
right.. simplify ;)